cookbooks/aws-parallelcluster-slurm/resources/munge_key_manager.rb

```diff
@@ -23,35 +23,43 @@
 
 default_action :setup_munge_key
 
-def fetch_and_decode_munge_key(munge_key_secret_arn)
-  declare_resource(:bash, 'fetch_and_decode_munge_key') do
+def restart_munge_service
+  declare_resource(:service, "munge") do
+    supports restart: true
+    action :restart
+    retries 5
+    retry_delay 10
+  end
+end
+
+def share_munge_key_to_dir(shared_dir)
+  declare_resource(:bash, 'share_munge_key') do
     user 'root'
     group 'root'
-    cwd '/tmp'
-    code <<-FETCH_AND_DECODE
+    code <<-SHARE_MUNGE_KEY
       set -e
-      # Get encoded munge key from secrets manager
-      encoded_key=$(aws secretsmanager get-secret-value --secret-id #{munge_key_secret_arn} --query 'SecretString' --output text --region #{node['cluster']['region']})
-      # If encoded_key doesn't have a value, error and exit
-      if [ -z "$encoded_key" ]; then
-        echo "Error fetching munge key from Secrets Manager or the key is empty"
-        exit 1
-      fi
+      mkdir -p #{shared_dir}/.munge
+      # Copy key to shared dir
+      cp /etc/munge/munge.key #{shared_dir}/.munge/.munge.key
+      chmod 0700 #{shared_dir}/.munge
+      chmod 0600 #{shared_dir}/.munge/.munge.key
+    SHARE_MUNGE_KEY
+  end
+end
 
-      # Decode munge key and write to /etc/munge/munge.key
-      decoded_key=$(echo $encoded_key | base64 -d)
-      if [ $? -ne 0 ]; then
-        echo "Error decoding the munge key with base64"
-        exit 1
-      fi
+def share_munge_head_node
+  share_munge_key_to_dir(node['cluster']['shared_dir'])
+  share_munge_key_to_dir(node['cluster']['shared_dir_login'])
```

Comment:
Let's check the use of `node['cluster']['shared_dir_login_nodes']` instead of `node['cluster']['shared_dir_login']` here.

`node['cluster']['shared_dir_login']` is set in the environment cookbook but we should probably rely on variables set in the `aws-parallelcluster-shared` which all cookbooks depend on.

Reply:
Done! Also changed in other places.